manifests/base.pp

```diff
@@ -44,27 +36,17 @@
   }
 
   if $rsyslog::manage_confdir {
-    $purge_params = $rsyslog::purge_config_files ? {
-      true  => {
-        'purge'   => true,
-        'recurse' => true,
-      },
-      false => {}
-    }
-
-    $require_package = $rsyslog::manage_package ? {
-      true => {
-        'require' => Package[$rsyslog::package_name],
-      },
-      false => {}
+    file { $rsyslog::confdir:
+      ensure  => directory,
+      owner   => 'root',
+      group   => 'root',
+      mode    => $rsyslog::confdir_permissions,
+      purge   => $rsyslog::purge_config_files,
+      recurse => $rsyslog::purge_config_files,
     }
 
-    file { $rsyslog::confdir:
-      ensure => directory,
-      owner  => 'root',
-      group  => 'root',
-      mode   => $rsyslog::confdir_permissions,
-      *      => $purge_params + $require_package,
+    if $rsyslog::manage_package {
+      Package[$rsyslog::package_name] -> File[$rsyslog::confdir]
```

Comment:
I see there was some discussion about this vs meta parameters. You can consider the following:
```puppet
$package_require = $rsyslog::manage_package ? {
  true => Package[$rsyslog::package_name],
  default => undef,
}

file { $file:
  require => $package_require,
}
```

I'm not sure if it's better. If there's one case, I'd say it's not. In this case there are 2 files using it which makes a better case.

Both are certainly better than what exists now with `* => $require_package`. That's just <del>magic</del> hard to follow and thus easy to miss.